core/src/main/java/org/apache/carbondata/core/metadata/SegmentFileStore.java

```diff
@@ -230,23 +231,36 @@ public static boolean writeSegmentFile(CarbonTable carbonTable, Segment segment)
     return false;
   }
 
-  public static boolean writeSegmentFileForOthers(CarbonTable carbonTable, Segment segment)
+  public static boolean writeSegmentFileForOthers(
+      CarbonTable carbonTable,
+      Segment segment,
+      PartitionSpec partitionSpec,
+      List<FileStatus> partitionDataFiles)
       throws IOException {
     String tablePath = carbonTable.getTablePath();
-    CarbonFile segmentFolder = FileFactory.getCarbonFile(segment.getSegmentPath());
-    CarbonFile[] otherFiles = segmentFolder.listFiles(new CarbonFileFilter() {
-      @Override public boolean accept(CarbonFile file) {
-        return (!file.getName().equals("_SUCCESS") && !file.getName().endsWith(".crc"));
-      }
-    });
-    if (otherFiles != null && otherFiles.length > 0) {
+    CarbonFile[] dataFiles = null;
+    if (partitionDataFiles.isEmpty()) {
+      CarbonFile segmentFolder = FileFactory.getCarbonFile(segment.getSegmentPath());
+      dataFiles = segmentFolder.listFiles(
+          file -> (!file.getName().equals("_SUCCESS") && !file.getName().endsWith(".crc")));
```

Comment:
dont use lamda.. some users might be using java7

Reply:
For CarbonData 2.0, we should change to use java 1.8, and this does not impact user code in client side

Reply:
Then let us change the java version for all the modules. Can you handle in this PR?